src/utils/applyWalletActions.js

```diff
@@ -0,0 +1,34 @@
+import Backend from './backend';
+import { createDeepLinkUrl } from './util';
+
+export const applyBackendChanges = async (method, request, requestParams, cb) => {
+  const args = {
+    sendProfileData: [...requestParams, request],
+    pinItem: [...requestParams, request],
+    unPinItem: [...requestParams, request],
+    sendPostReport: [request],
+  }[method];
+  if (!args) throw new Error(`Unknown method: ${method}`);
+  await Backend[method](...args);
+  if (cb) cb();
+};
+
+export const backendAuth = async (method, challenge, client, requestParams, url, cb) => {
```

Comment:
we can make these methods as a mixin, to access `sdk` from `this`

Reply:
This would require further refactoring of how client object is initialized and retrieved. 
Here we have two cases:
- with established rpc client connection
- no client connection 
Both are covered so far.
Do you think we can merge this PR and think of restructuring the client initialization?

Reply:
actually, I have a refactored version, but it depends on #708